### PR TITLE
Cache JSX transformations

### PR DESCRIPTION
**Pull request by intrnl**
Improves subsequent JSX requires by [up to 48 percent (personal test, may vary)](https://haste.powercord.dev/muhapohove.txt)

### CODE REVIEW

**Comment by intrnl** [comment]
This caching implementation does not have a mapping of file names to their respective cache files, so this will be painful when developing as suddenly there would be a lot of trash generated.

Since I don't think adding mappings would add any performance penalties, I'm thinking of baking that in right now. 👍

**Comment by intrnl** [comment]
Welp, I don't think having mappings is worth the effort considering the edge cases that comes along with it. So I'd say this is good to go.

**Comment by intrnl** [comment]
Well it seems that the CI is failing thanks to an unrelated file

**Comment by Zerthox** [comment]
The additional generated trash during development should be pretty small, since the only scenarios where it'll transform the same content is when the file has been modified but the contents haven't changed or have been reverted to an already cached state.

For further trash reduction the following ideas might be worth considering:
1. Make the cache clean itself. Remove already cached instances of a file on transform. This would also remove old files when Powercord updates, but it requires mappings.
2. Add an option to disable caching. This might be worth implementing for users who are short on disk space aswell.
3. Add a command to clear (or clean) the cache. This would be a little annoying to use during development, but it would also give end users more control over the cache.

EDIT: @intrnl you forgot to revert the cache name in the gitignore aswell

File: src/Powercord/modules/jsx.js [hunk 1]
```diff
@@ -17,14 +17,40 @@
...
-const { readFileSync } = require('fs');
+const { join } = require('path');
+const { readFileSync, existsSync, mkdirSync, writeFile } = require('fs');
+const { createHash } = require('crypto');
 
-module.exports = () =>
-  require.extensions['.jsx'] = (_module, filename) =>
-    _module._compile(
-      sucrase.transform(readFileSync(filename, 'utf8'), {
+const cacheDir = join(__dirname, '../../../cache/jsx/');
```
Comment (cyyynthia) [inline src/Powercord/modules/jsx.js]: gitignore mentions `.cache` but you use `cache`
```suggestion
const cacheDir = join(__dirname, '../../../.cache/jsx/');
```

File: src/Powercord/modules/jsx.js [hunk 2]
```diff
@@ -17,14 +17,40 @@
...
-      }).code,
-      filename
-    );
+      }).code;
+
+      _module._compile(res, filename);
+
+      writeFile(cached, res, (err) => {
+        if (err) {
+          console.error('[JSX]', 'Failed to write to cache');
```
Comment (cyyynthia) [inline src/Powercord/modules/jsx.js]: Pretty:tm: logging
```suggestion
          console.error('%c[Powercord:JSX]', 'color: #7289da', 'Failed to write to cache');
```

**Review by cyyynthia — CHANGES_REQUESTED** [review]

**Review by cyyynthia — APPROVED** [review]

**Pull request merged**